### PR TITLE
feat: Display dcl and ens worlds in worlds view

### LINKED ISSUE #2913: Create table to list DCL world deployments of a user

### LINKED ISSUE #2914: Create table to list ENS world deployments of a user

### LINKED ISSUE #2931: Display the ENS worlds list view or DCL worlds list view depending on the selected tab

### PR DESCRIPTION
**Pull request by fzavalia**
Closes #2931
Closes #2913
Closes #2914

image

image

### CODE REVIEW

File: src/components/WorldListPage_WorldsForEnsOwnersFeature/utils.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const fromBytesToMegabytes = (bytes: number) => {
+  return bytes / 1000000
```
Comment (fzavalia) [inline src/components/WorldListPage_WorldsForEnsOwnersFeature/utils.ts:2]: https://github.com/decentraland/builder/compare/feat/blocked-state#diff-0f09525ed78eee410b675c0f050c346cf3bb2245e384a1c1fc78637703870afeR2 

Updated to this in a future PR

```
return bytes / 1024 / 1024
```

File: src/modules/ens/utils.spec.ts [hunk 2]
```diff
@@ -23,3 +26,64 @@ describe('when checking if a subdomain is an external subdomain', () => {
...
+            scenesUrn: [
+              'urn:decentraland:entity:bafkreiavjamezigndx7rq5ggwd3tg3c5hzrnvla5euukqs25ew2hs4gtte?=&baseUrl=https://worlds-content-server.decentraland.zone/contents/'
+            ]
+          }
+        } as WorldInfo
+
+        jest.spyOn(content, 'fetchWorld').mockResolvedValue(response)
+      })
+
+      it('should return the enss with the world status set to null', async () => {
```
Comment (LautaroPetaccio) [inline src/modules/ens/utils.spec.ts:70]:
```suggestion
      it('should return the enss with the world status set', async () => {
```

File: src/modules/ens/selectors.ts [hunk 3]
```diff
@@ -42,6 +42,11 @@ export const getExternalNamesForConnectedWallet = createSelector(getExternalName
   return Object.values(externalNames).filter(externalName => isEqual(externalName.nftOwnerAddress, address))
 })
 
+export const getExternalNamesForWallet = (wallet: string) => (state: RootState) => {
```
Comment (LautaroPetaccio) [inline src/modules/ens/selectors.ts:45]: Would you mind adding tests for this new selector?

**Review by LautaroPetaccio — COMMENTED** [review]
The PR looks great! 👍  I've left a few comments to discuss / fix

File: src/modules/worlds/sagas.ts [hunk 4]
```diff
@@ -1,53 +1,26 @@
...
- * This saga is intended to be used without providing an address only if a wallet is connected or connecting.
- * If called without a connected wallet and providing no address, it will get stuck until a wallet is connected.
- */
 function* handlefetchWorldsWalletStatsRequest(action: FetchWalletWorldsStatsRequestAction) {
-  let address = action.payload.address
-
-  if (!address) {
-    address = yield select(getAddress)
-  }
+  const address = action.payload.address ?? (yield call(getAddressOrWaitConnection))
```
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/modules/worlds/sagas.ts]: Removed it as it is not necessary anymore given that the request is made when the wallet connects.

File: src/modules/translation/languages/en.json [hunk 5]
```diff
@@ -601,7 +601,8 @@
       "status_active": "Active",
       "status_inactive": "Inactive",
       "actions": "Actions",
-      "empty_url": "To activate this world you need to publish a scene"
+      "empty_url": "To activate this world you need to publish a scene",
+      "size": "Size mb"
```
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/modules/translation/languages/en.json]: Replaced all mb occurrences with Mb.

File: src/components/WorldListPage_WorldsForEnsOwnersFeature/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/components/WorldListPage_WorldsForEnsOwnersFeature/utils.ts:2]: Doing so in another PR https://github.com/decentraland/builder/pull/2936/files#diff-0f09525ed78eee410b675c0f050c346cf3bb2245e384a1c1fc78637703870afeR4

File: src/modules/ens/selectors.ts (hunk 3 above)
Comment (fzavalia) [reply src/modules/ens/selectors.ts:45]: Added :D

**Review by LautaroPetaccio — APPROVED** [review]
Send it 🚀

**Pull request merged**